### DISCUSSION TITLE
Wrapping Java jar as service

### DISCUSSION THREAD

**Discussion by lazamius** [kohsuke/winsw#892] [Q&A] [answered]
My Java program grabs screenshots using Robot class and sends them over socket. It works fine until I wrap it with winsw and run it as a service. Then all I get on the receiving end is black image, no actual screenshots. 
Can Robot take screenshots from a service, or is the problem somewhere else?

**Comment by John-Chan** [comment, marked as answer]
Services cannot directly interact with user 

- https://docs.microsoft.com/en-us/windows/win32/services/interactive-services
- https://serverfault.com/a/576178

**Reply by lazamius** [reply]
Thanks.
